File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/dependencies/SimpleDependency_ViewModelSpy.cs
/// <filename>
///     SimpleDependency_ViewModelSpy.cs
/// </filename>
using System.Collections.Generic;
using dependencies;

namespace dependencies
{
    public class SimpleDependency_ViewModelSpy : dependencies.SimpleDependency_ViewModel
    {
        private dependencies.SimpleDependency_ViewModel wrapped;

        public virtual void setWrapped(dependencies.SimpleDependency_ViewModel wrappedObject)
        {
            this.wrapped = wrappedObject;
        }

        public struct LoadViewModelCallInfo
        {
            public string PathValue;

            public LoadViewModelCallInfo()
            {
            }

            public LoadViewModelCallInfo(string PathValue)
            {
                this.PathValue = PathValue;
            }
        }

        public System.Collections.Generic.List<dependencies.SimpleDependency_ViewModelSpy.LoadViewModelCallInfo> LoadViewModelCallInfos = new System.Collections.Generic.List<dependencies.SimpleDependency_ViewModelSpy.LoadViewModelCallInfo>();

        public virtual void LoadViewModel(string Path)
        {
            dependencies.SimpleDependency_ViewModelSpy.LoadViewModelCallInfo callInfo = new dependencies.SimpleDependency_ViewModelSpy.LoadViewModelCallInfo();
            callInfo.PathValue = Path;
            this.LoadViewModelCallInfos.Add(callInfo);
            if (this.wrapped != null)
            {
                this.wrapped.loadViewModel(Path);
            }
        }
    }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/dependencies/SimpleDependency_ViewModelSpy.cpp
#include "SimpleDependency_ViewModelSpy.h"
#include <vector>
#include "SimpleDependency_ViewModel.h"
#include <memory>
#include <string>

namespace dependencies
{
  void SimpleDependency_ViewModelSpy::setWrapped(std::shared_ptr<SimpleDependency_ViewModel> wrappedObject)
  {
    this->wrapped = wrappedObject;
  }
  void SimpleDependency_ViewModelSpy::LoadViewModel(std::string Path)
  {
    SimpleDependency_ViewModelSpy::LoadViewModelCallInfo callInfo = { };
    callInfo.PathValue = Path;
    this->LoadViewModelCallInfos.push_back(callInfo);
    if (this->wrapped != nullptr)
    {
      this->wrapped->loadViewModel(Path);
    }
  }
}
